### ISSUE TITLE
How to use the validate_value method

### ISSUE THREAD

**Issue by behaba** [devgeniem/acf-codifier#40]
Hi !

I'm trying to use the `validate_value` method but I don't understand how it works.

I declare the field like this :

```
$phone_fld = new Field\Text( __( 'Phone' ), 'phone', 'phone'  );
$phone_fld->validate_value( array( '\Validator', 'phone' ) );
```

And the validator is called like this :

```
class Validator {
	public function phone( $str ) {
		return preg_match( '/^(0|\+33)[1-9]([-. ]?[0-9]{2}){4}$/', $str );
	}
}
 ```

the problem is that I can not get the value of the field ($ str).

Is this the right way to proceed?

Thanks for your help.

**Comment by Nomafin** [comment]
Hi, ACF's native `validate_value` takes four parameters: `$valid`, `$value`, `$field` and `$input`. Codifier's `validate_value()` is just a proxy for the original function, so you need to have same parameters in your function.

More info from ACF documentation: https://www.advancedcustomfields.com/resources/acf-validate_value/

**Comment by behaba** [comment]
I think there is a problem with the arguments.

The current filter only return one argument (accepted args by default).

I think it could be great to set the missing priority and accepted args like this :
```
$this->filters['validate_value'] = [
            'filter'   => 'acf/validate_value/key=',
            'function' => $function,
            'priority' => 10,
            'accepted_args' => 4
];
```

**Comment by behaba** [comment]
I submitted a pull request which résolve the problem

**Comment by Nomafin** [comment]
The pull request was approved and released with 1.10.0.
